apps/labs/posts/cupy-czi-grant-year1.md

```diff
@@ -0,0 +1,121 @@
+
+During the first year of the grant work, a team of developers at Quansight Labs have teamed up with CuPy maintainers at Preferred Networks to increase the coverage of interpolation and signal processing subpackages, where, as of CuPy pre-release v13.0-b1, 18 interpolation and more than 100 signal APIs are now available, including basis splines, regular grid interpolation, and piecewise polynomials for the interpolation module and IIR linear filtering, filter design, linear time invariant systems, and peak finding for the signal processing subpackage.
+
+As expected, all the ported APIs afford a significant performance increase relative to the SciPy CPU reference implementations by exploiting the parallel nature of GPU kernels against the sequential nature of CPU implementations. For example, the new CuPy `BSpline` implementation has a nearly 10x performance improvement when compared to its corresponding implementation in SciPy, and `PPoly` has a 32x performance improvement, representing a huge enhancement over CPU implementations.
+
+
+<figure>
+  <img src="/posts/cupy-czi-grant-year1/interp_performance.png" alt="A performance comparison graph of three CuPy interpolation APIs compared against their corresponding SciPy counterparts, across different array input sizes. The performance of CuPy exceeds SciPy by almost 30 times for large inputs." />
+  <figcaption style={{textAlign: "center"}}>Relative speedup of CuPy’s <tt>BSpline</tt>, <tt>PPoly</tt> and <tt>BPoly</tt> interpolation classes relative to SciPy for various input sizes.</figcaption>
+</figure>
+
+For those APIs which could not be readily ported from CPU to GPU, new implementations were derived from scientific papers, including the new IIR linear filtering routines, which represent the first time IIR filters can be applied on GPUs from Python. As part of this process, some enhancements and simplifications were also backported to SciPy, such as a major refactor of symmetric IIR filters to leverage IIR linear filtering and second order sections rather than dedicated routines that reduplicate code.
+
+The following example demonstrates the new capabilities of CuPy in filtering multiple signals in parallel using a GPU. For illustrative purposes, we begin by defining four different signals.
+
+```python
+import cupy as cp
+from cupyx.scipy import signal
+import matplotlib.pyplot as plt
+
+cp.random.seed(2354)
+
+rng = cp.random.default_rng(seed=1234)
```

Comment:
Is it necessary to set two different seeds here?

Reply:
If I used the same seed in both the global and the particular generators, the signals produced don't have much variety, hence the two seeds